runtime/permissions/lib.rs

```diff
@@ -1135,15 +1166,18 @@ impl QueryDescriptor for EnvDescriptor {
   }
 
   fn from_allow(allow: &Self::AllowDesc) -> Self {
-    allow.clone()
+    match allow {
+      Self::AllowDesc::Name(s) => Self(s.clone()),
+      Self::AllowDesc::PrefixPattern(_s) => todo!(),
```

Comment:
Not sure how to handle this one :/

Reply:
I think you'll need to add a way to represent this in the EnvQueryDescriptor, but it's only used for this purposes. We just need to ensure that where this is used works appropriately.